// @superwall/react-native-superwall/lib/module/public/LogScope.js
export let LogScope = /*#__PURE__*/function (LogScope) {
  LogScope["LocalizationManager"] = "localizationManager";
  LogScope["BounceButton"] = "bounceButton";
  LogScope["CoreData"] = "coreData";
  LogScope["ConfigManager"] = "configManager";
  LogScope["IdentityManager"] = "identityManager";
  LogScope["DebugManager"] = "debugManager";
  LogScope["DebugViewController"] = "debugViewController";
  LogScope["LocalizationViewController"] = "localizationViewController";
  LogScope["GameControllerManager"] = "gameControllerManager";
  LogScope["Device"] = "device";
  LogScope["Network"] = "network";
  LogScope["PaywallEvents"] = "paywallEvents";
  LogScope["ProductsManager"] = "productsManager";
  LogScope["StoreKitManager"] = "storeKitManager";
  LogScope["Placements"] = "placements";
  LogScope["Receipts"] = "receipts";
  LogScope["SuperwallCore"] = "superwallCore";
  LogScope["PaywallPresentation"] = "paywallPresentation";
  LogScope["PaywallTransactions"] = "paywallTransactions";
  LogScope["PaywallViewController"] = "paywallViewController";
  LogScope["Cache"] = "cache";
  LogScope["All"] = "all";
  return LogScope;
}({});
